### ISSUE TITLE
snmp.createReceiver (Error: bind EACCES 0.0.0.0:162)

### ISSUE THREAD

**Issue by Ekristoffe** [markabrahams/node-net-snmp#86]
Hello,
I am working on Node-RED inside a docker app.
I know about the fact I could run Node-RED in root to solve (hide) this problem but I would like to have a clean solution to it.
If I call this function:
```
receiver = snmp.createReceiver (options, callback);
```
with a normal user I have this error:
```
7 May 15:23:06 - [info] Starting flows
7 May 15:23:07 - [log] createReceiver :  { port: 162,
  disableAuthorization: true,
  engineID: '8000B98380XXXXXXXXXXXX',
  transport: 'udp4' }
7 May 15:23:07 - [info] Started flows
7 May 15:23:07 - [red] Uncaught Exception:
7 May 15:23:07 - Error: bind EACCES 0.0.0.0:162
    at state.handle.lookup (dgram.js:242:18)
    at process._tickCallback (internal/process/next_tick.js:63:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-red-docker@1.0.6 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
npm ERR! Exit status 1
npm ERR! Failed at the node-red-docker@1.0.6 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /usr/src/node-red/.npm/_logs/2020-05-07T15_23_07_733Z-debug.log
```
And here is the detailed log:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'start',
1 verbose cli   '--',
1 verbose cli   '--userDir',
1 verbose cli   '/data' ]
2 info using npm@6.14.4
3 info using node@v10.20.1
4 verbose config Skipping project config: /usr/src/node-red/.npmrc. (matches userconfig)
5 verbose run-script [ 'prestart', 'start', 'poststart' ]
6 info lifecycle node-red-docker@1.0.6~prestart: node-red-docker@1.0.6
7 info lifecycle node-red-docker@1.0.6~start: node-red-docker@1.0.6
8 verbose lifecycle node-red-docker@1.0.6~start: unsafe-perm in lifecycle true
9 verbose lifecycle node-red-docker@1.0.6~start: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/usr/src/node-red/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
10 verbose lifecycle node-red-docker@1.0.6~start: CWD: /usr/src/node-red
11 silly lifecycle node-red-docker@1.0.6~start: Args: [ '-c',
11 silly lifecycle   'node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"' ]
12 silly lifecycle node-red-docker@1.0.6~start: Returned: code: 1  signal: null
13 info lifecycle node-red-docker@1.0.6~start: Failed to exec start script
14 verbose stack Error: node-red-docker@1.0.6 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
14 verbose stack     at EventEmitter.emit (events.js:198:13)
14 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
14 verbose stack     at ChildProcess.emit (events.js:198:13)
14 verbose stack     at maybeClose (internal/child_process.js:982:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
15 verbose pkgid node-red-docker@1.0.6
16 verbose cwd /usr/src/node-red
17 verbose Linux 4.9.115-rt93
18 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "start" "--" "--userDir" "/data"
19 verbose node v10.20.1
20 verbose npm  v6.14.4
21 error code ELIFECYCLE
22 error errno 1
23 error node-red-docker@1.0.6 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
23 error Exit status 1
24 error Failed at the node-red-docker@1.0.6 start script.
24 error This is probably not a problem with npm. There is likely additional logging output above.
25 verbose exit [ 1, true ]
```
How could I catch the error instead of crashing the whole application ?
I tried a try/catch but i already know it doesn't work with async function.
If anyone have a good solution (also maybe update the example too).

Thanks.

**Comment by markabrahams** [comment]
Hi @Ekristoffe - I've changed the behaviour of the library from throwing an error on this condition (current behaviour) to invoking the supplied callback function with the Error object.

This is released to master and published in version 2.6.4 of the npm.

**Comment by Ekristoffe** [comment]
it work like a charm.
than you.
